urllib3/connection.py

```diff
@@ -133,11 +134,15 @@ def _new_conn(self):
             conn = connection.create_connection(
                 (self.host, self.port), self.timeout, **extra_kw)
 
-        except SocketTimeout:
+        except SocketTimeout as e:
             raise ConnectTimeoutError(
                 self, "Connection to %s timed out. (connect timeout=%s)" %
                 (self.host, self.timeout))
 
+        except SocketError as e:
+            raise NewConnectionError(
+                self, "Failed to establish a new connection: %s" % e)
```

Comment:
Maybe, but I don't think I like the name (maximum bikeshedding!). `ConnectionSetupError`?